test/client-test.js

```diff
@@ -71,6 +71,39 @@ suite('client-test', () => {
     });
   });
 
+  test('it instantiates a GraphQL client with the given config and language header when a language config is provided', () => {
+    let passedTypeBundle;
+    let passedUrl;
+    let passedFetcherOptions;
+
+    class FakeGraphQLJSClient {
+      constructor(typeBundle, {url, fetcherOptions}) {
+        passedTypeBundle = typeBundle;
+        passedUrl = url;
+        passedFetcherOptions = fetcherOptions;
+      }
+    }
+
+    const withLanguageConfig = {
+      domain: 'sendmecats.myshopify.com',
+      storefrontAccessToken: 'abc123',
+      language: 'ja-JP'
+    };
+
+    new Client(new Config(withLanguageConfig), FakeGraphQLJSClient); // eslint-disable-line no-new
+
+    assert.deepEqual(passedTypeBundle, types);
+    assert.equal(passedUrl, `https://sendmecats.myshopify.com/api/${config.apiVersion}/graphql`);
+    assert.deepEqual(passedFetcherOptions, {
+      headers: {
+        'X-SDK-Variant': 'javascript',
+        'X-SDK-Version': version,
+        'X-Shopify-Storefront-Access-Token': config.storefrontAccessToken,
```

Comment:
Should we be asserting against `withLanguageConfig` instead of `config`?

Reply:
Taking another look at this, I refactored the few tests to be a bit more concise and scoped.